### PR TITLE
Add Components V2 route shell

### PR DESCRIPTION
**Pull request by Mbeaulne**
## Tophatting

Manual tophatting recommended because this adds a new route.

1. Turn on the Components V2 beta flag.
2. Confirm the dashboard sidebar shows “Components V2”.
3. Open `/components-v2`.
4. Turn the flag off and confirm direct navigation redirects away.

## What changed

Adds the first shell of the experimental Components V2 page.

This PR adds:

- the `component-search-v2` feature flag
- the `/components-v2` route
- the dashboard sidebar link when the flag is on
- the basic page header and search input

It does not add real search results yet. That comes in the next PR.

## Why

This keeps routing and feature-flag review separate from the larger search implementation.

## Test plan

- Turn on the Components V2 beta flag
- Confirm the dashboard sidebar shows “Components V2”
- Open `/components-v2`
- Turn the flag off and confirm direct navigation redirects away
- Run `pnpm run typecheck --pretty false`

### CODE REVIEW

**Comment by Mbeaulne** [comment]
* **#2337** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2337?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#2350](https://github.com/TangleML/tangle-ui/pull/2350) <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2350?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2351](https://github.com/TangleML/tangle-ui/pull/2351) <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2351?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#2323** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2323?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2322** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2322?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2319** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2319?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2319?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#2318** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2318?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2317** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2317?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2321** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2321?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2320** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2320?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2316** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2316?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2315** <a href="https://app.graphite.com/github/pr/TangleML/tangle-ui/2315?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by morgan-wowk — APPROVED** [review]

File: src/routes/Dashboard/DashboardComponentsV2View.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+export const DashboardComponentsV2View = () => {
+  const [query, setQuery] = useState("");
+
+  const handleQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  return (
+    <div
```
Comment (camielvs) [inline src/routes/Dashboard/DashboardComponentsV2View.tsx:23]: > 🤖 This is an AI-generated code review comment.

**Use `BlockStack` instead of this raw `flex flex-col` div.**

```tsx
<div
  className="flex flex-col -mt-4 -mb-6 -mx-8 overflow-hidden"
  style={{ height: `calc(100vh - ${TOP_NAV_HEIGHT}px)` }}
>
```

This is a vertical flex container — exactly what `BlockStack` provides. The children's `shrink-0` / `flex-1 min-h-0` still work since `BlockStack` renders a `flex-col` div.

Fix: `<BlockStack className="-mt-4 -mb-6 -mx-8 overflow-hidden" style={{ height: … }}>` (set `gap` explicitly — none is wanted here since the header carries its own `border-b`). The two inner non-flex padded/scroll `<div>`s are fine as raw divs.

File: src/routes/Dashboard/DashboardComponentsV2View.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+              Search across component sources from one experimental dashboard.
+            </Paragraph>
+          </BlockStack>
+          <Input
+            type="search"
+            placeholder="e.g. train_test_split, pandas, clean up my data"
+            value={query}
+            onChange={handleQueryChange}
+            aria-label="Search components"
+            className="flex-1"
```
Comment (camielvs) [inline src/routes/Dashboard/DashboardComponentsV2View.tsx]: > 🤖 This is an AI-generated code review comment.

**`flex-1` on the search `Input` is a no-op leftover here.**

The `Input` sits in a `BlockStack` (`flex-col`) with `align="stretch"`, so it is already full-width; `flex-1` acts on the *vertical* main axis and does nothing useful. This pattern makes sense in `AgentSettings`, where the `Input` was in a horizontal `InlineStack` — looks copy-pasted from there.

Fix: drop `className="flex-1"`.

File: src/routes/Dashboard/DashboardLayout.tsx [hunk 3]
```diff
@@ -41,6 +48,18 @@ const navItemClass = (isActive: boolean) =>
...
 export function DashboardLayout() {
   const requiresAuthorization = isAuthorizationRequired();
+  const isComponentsV2Enabled = useFlagValue("component-search-v2");
+
+  // Insert the Components V2 entry directly after "Components" when the
+  // beta flag is on. Keeps the nav order intuitive without touching the
+  // base list.
+  const sidebarItems = isComponentsV2Enabled
+    ? [
+        ...BASE_SIDEBAR_ITEMS.slice(0, 4),
```
Comment (camielvs) [inline src/routes/Dashboard/DashboardLayout.tsx]: > 🤖 This is an AI-generated code review comment.

**Magic-number slice for the nav insertion is fragile.**

```tsx
const sidebarItems = isComponentsV2Enabled
  ? [...BASE_SIDEBAR_ITEMS.slice(0, 4), COMPONENTS_V2_ITEM, ...BASE_SIDEBAR_ITEMS.slice(4)]
  : BASE_SIDEBAR_ITEMS;
```

This is correct today — `"Components"` is at index 3, so V2 lands right after it. But the `4` is positional magic tied to the exact array order; reordering or inserting a `BASE_SIDEBAR_ITEMS` entry would silently misplace the V2 item, and the comment ("directly after Components") states a semantic intent the code does not enforce.

Fix: derive the index —
```tsx
const idx = BASE_SIDEBAR_ITEMS.findIndex((i) => i.to === APP_ROUTES.DASHBOARD_COMPONENTS) + 1;
const sidebarItems = isComponentsV2Enabled
  ? [...BASE_SIDEBAR_ITEMS.slice(0, idx), COMPONENTS_V2_ITEM, ...BASE_SIDEBAR_ITEMS.slice(idx)]
  : BASE_SIDEBAR_ITEMS;
```

**Review by camielvs — APPROVED** [review]

File: src/routes/Dashboard/DashboardComponentsV2View.tsx (hunk 1 above)
Comment (Mbeaulne) [reply src/routes/Dashboard/DashboardComponentsV2View.tsx:23]: > 🤖 This is an AI-generated response.

Fixed — the vertical-flex semantic is now on a `BlockStack`. The outer `<div>` is kept around it solely to carry the inline `style={{ height: calc(...) }}` because `BlockStack`'s typed props don't include `style` (`AriaAttributes & VariantProps & { as, fill, className }`). The two inner padded/scroll `<div>`s stay as raw divs since they aren't vertical flex containers.

Same constraint hit on PR #2317 — happy to add `style` to `BlockStackProps` as a follow-up if we want to drop the wrapper here and everywhere similar.

File: src/routes/Dashboard/DashboardComponentsV2View.tsx (hunk 2 above)
Comment (Mbeaulne) [reply src/routes/Dashboard/DashboardComponentsV2View.tsx]: > 🤖 This is an AI-generated response.

Fixed — dropped `className="flex-1"`. You're right, it was a copy-paste from `AgentSettings` where the surrounding container was horizontal.

File: src/routes/Dashboard/DashboardLayout.tsx (hunk 3 above)
Comment (Mbeaulne) [reply src/routes/Dashboard/DashboardLayout.tsx]: > 🤖 This is an AI-generated response.

Fixed — now derives the index via `findIndex(item => item.to === APP_ROUTES.DASHBOARD_COMPONENTS)` and inserts at `index + 1`. Falls back to appending at the end if the entry ever goes missing (defensive against a future refactor that drops the Components nav item but forgets to update this insertion). Comment also updated to reflect the new behaviour.

**Comment by Mbeaulne** [comment]
### Merge activity

* **Jun 5, 2:37 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/TangleML/tangle-ui/2319).
* **Jun 5, 3:07 PM UTC**: [Graphite](https://app.graphite.com/github/pr/TangleML/tangle-ui/2319) couldn't merge this PR because it had merge conflicts.
* **Jun 5, 5:42 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/TangleML/tangle-ui/2319).
* **Jun 5, 5:42 PM UTC**: @Mbeaulne merged this pull request with [Graphite](https://app.graphite.com/github/pr/TangleML/tangle-ui/2319).
